### DISCUSSION TITLE
Possibility in plugin framework to generate Workspaces/Request collections

### DISCUSSION THREAD

**Discussion by matthijssn** [Kong/insomnia#3840] [Ideas]
A possibility to create workspaces/request collections (including folders and environments) to be able to create custom wizards to define project structures.

**Comment by dimitropoulos** [comment]
Interesting!  Can you say more about the plugin you're thinking of making?  What's the underlying use-case?

**Reply by matthijssn** [reply]
Hi Dimitro, thanks for replying. I was thinking about a way to create wizards as plugins for generating workspace structures. In my team we integrate our ERP solution with the customers application landscape. It would be nice if I can create a wizard for my team which asks the standard questions like (examples:) what's the customers endpoints? what's the oauth2 info?, do they have [insert standard application name]? Do they have a Development/Test/Acceptance/Productions environment? Based on those questions the plugin can generate a standard, customer specific workspace with a default set of environment settings based on their own DTAP and a set of specific requests based on our best practice. Hope this makes sense.

**Reply by wdawson** [reply]
This is really cool! We definitely want to expand the surface area of Insomnia that plugins can interact with. This seems like a good area to explore. We'll continue monitoring this discussion for interest but would also welcome a PR to try to do this.
